tests/test_storm_control.py

```diff
@@ -0,0 +1,528 @@
+        assert status == True
+        print ("\nPort field value pairs before delete")
+        print (fvs)
+
+        policer_oid = 0
+
+        for fv in fvs:
+            if fv[0] == "SAI_PORT_ATTR_BROADCAST_STORM_CONTROL_POLICER_ID":
+                policer_oid = fv[1]
+
+        ###### DELETE ######
+
+        self.delete_storm_session(if_name, storm_type)
+        tbl = swsscommon.Table(self.cdb,"PORT_STORM_CONTROL")
+        (status,fvs) = tbl.get(if_name+"|"+storm_type)
+
+        assert status == False
+
+        atbl = swsscommon.Table(self.adb,"ASIC_STATE:SAI_OBJECT_TYPE_PORT")
+        status, fvs = atbl.get(dvs.asicdb.portnamemap[if_name])
+        assert status == True
+        print ("Port field value pairs after delete")
+        print (fvs)
+
+        for fv in fvs:
+            if fv[0] == "SAI_PORT_ATTR_BROADCAST_STORM_CONTROL_POLICER_ID":
+                assert fv[1] == "oid:0x0"
+
+        if policer_oid != 0:
+            atbl = swsscommon.Table(self.adb,"ASIC_STATE:SAI_OBJECT_TYPE_POLICER")
+            status, fvs = atbl.get(policer_oid)
+            assert status == False
+
+    def test_del_uucast_storm(self,dvs,testlog):
+        #Proceeding with assumption that storm-control is already enabled on interface
+        self.setup_db(dvs)
+
+        if_name = "Ethernet0"
+        storm_type = "unknown-unicast"
+
+        port_oid = dvs.asicdb.portnamemap[if_name]
+
+        atbl = swsscommon.Table(self.adb,"ASIC_STATE:SAI_OBJECT_TYPE_PORT")
+        status, fvs = atbl.get(dvs.asicdb.portnamemap[if_name])
+        assert status == True
+        print ("\nport field value pairs before delete")
+        print (fvs)
+
+        policer_oid = 0
+
+        for fv in fvs:
+            if fv[0] == "SAI_PORT_ATTR_FLOOD_STORM_CONTROL_POLICER_ID":
+                policer_oid = fv[1]
+
+        ###### DELETE ######
+
+        self.delete_storm_session(if_name, storm_type)
+        tbl = swsscommon.Table(self.cdb,"PORT_STORM_CONTROL")
+        (status,fvs) = tbl.get(if_name+"|"+storm_type)
+
+        assert status == False
+
+        atbl = swsscommon.Table(self.adb,"ASIC_STATE:SAI_OBJECT_TYPE_PORT")
+        status, fvs = atbl.get(dvs.asicdb.portnamemap[if_name])
+        assert status == True
+        print ("Port field value pairs after delete")
+        print (fvs)
+
+        for fv in fvs:
+            if fv[0] == "SAI_PORT_ATTR_FLOOD_STORM_CONTROL_POLICER_ID":
+                assert fv[1] == "oid:0x0"
+
+        if policer_oid != 0:
+            atbl = swsscommon.Table(self.adb,"ASIC_STATE:SAI_OBJECT_TYPE_POLICER")
+            status, fvs = atbl.get(policer_oid)
+            assert status == False
+
+    def test_del_umcast_storm(self,dvs,testlog):
+        #Proceeding with assumption that storm-control is already enabled on interface
+        self.setup_db(dvs)
+
+        if_name = "Ethernet0"
+        storm_type = "unknown-multicast"
+
+        port_oid = dvs.asicdb.portnamemap[if_name]
```

Comment:
replace the add part with call to add_storm_control_interface()

Reply:
The add part is based on existing storm-control configurations present on the interface.